### PR TITLE
MRG Default ica.plot_properties to a parametric CI

### LINKED ISSUE #6268: ENH: speed up ica.plot_properties
I currently have to wait some time after clicking a component in `ica.plot_components` window and the time seems to be mostly from waiting for the multitaper psd estimation to complete. I think that we should use something much simpler like welch or even fft as the default. The plotted spectrum is for informational reasons, not for analysis. I can compare timings on one of my datasets.

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Closes #6268

I am getting approx. half a second speedup for the tests on my machine.

Note I changed the functionality of plot_compare_evokeds a bit: now, as the doctstring already said, it will skip the CI if any condition has fewer than 2 evokeds.

@mmagnuski

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
I also added this to `plot_compare_evoked` with `axes='topo'`. This speeds up the plot by a lot, e.g. 4fold for 30 chaos! Not much for tests though.

File: mne/stats/parametric.py [hunk 1]
```diff
@@ -383,10 +383,4 @@ def _parametric_ci(arr, ci=.95):
...
     from scipy import stats
     sigma = stats.sem(arr, 0)
-    # This is highly convoluted to support 17th century Scipy
-    # XXX Fix when Scipy 0.12 support is dropped!
-    # then it becomes just:
-    # return stats.t.interval(ci, loc=mean, scale=sigma, df=arr.shape[0])
-    return np.asarray([stats.t.interval(ci, arr.shape[0],
-                       loc=mean_, scale=sigma_)
-                       for mean_, sigma_ in zip(mean, sigma)]).T
+    return stats.t.interval(ci, loc=mean, scale=sigma, df=arr.shape[0])
```
Comment (mmagnuski) [inline mne/stats/parametric.py]: are we out of the 17th century? ;)

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -2046,7 +2047,11 @@ def plot_compare_evokeds(evokeds, picks=None, gfp=False, colors=None,
         callable, it must take as its single argument an array
         (observations x times) and return the upper and lower confidence bands.
         If None or False, no confidence band is plotted.
-        If True, a 95%% bootstrapped confidence interval is drawn.
+        If True, a 95%% bootstrapped confidence interval is drawn, unless
+        ``evokeds`` does not have > 2 Evoked objects for each condition,
+        in which case no CI is drawn.
+        To plot a CI and a single condition, supply a list containing a list
+        containing multiple Evoked objects.
```
Comment (mmagnuski) [inline mne/viz/evoked.py]: add an example in parethesis so that it is super-clear: 
> (for example: `[[Evoked, Evoked, Evoked]]`)

**Comment by mmagnuski** [comment]
Looks good. How are the shades drawn for the psd? It's a different function for that, right?

**Comment by jona-sassenhagen** [comment]
Yeah and I think you wrote it :D

**Comment by mmagnuski** [comment]
That's right :) I was just trying to check whether psd stds were not refactored to be drawn via plot_compare_evokeds. I made sure by looking at the code now.
Then I don't have any other questions, LGTM. :)

**Comment by mmagnuski** [comment]
Travis and Azure complain.

File: mne/viz/evoked.py [hunk 3]
```diff
@@ -2046,7 +2047,12 @@ def plot_compare_evokeds(evokeds, picks=None, gfp=False, colors=None,
         callable, it must take as its single argument an array
         (observations x times) and return the upper and lower confidence bands.
         If None or False, no confidence band is plotted.
-        If True, a 95%% bootstrapped confidence interval is drawn.
+        If True, a 95%% bootstrapped confidence interval is drawn, unless
+        ``evokeds`` does not have > 2 Evoked objects for each condition,
+        in which case no CI is drawn.
+        To plot a CI and a single condition, supply a list containing a list
+        containing multiple Evoked objects (like ``[[Evoked1, Evoked2,
+        Evoked3]]``).
```
Comment (larsoner) [inline mne/viz/evoked.py]: The complaint is about this docstring getting to be too long. Can some of this be moved to a `Notes` section?

**Comment by jona-sassenhagen** [comment]
Done I think

File: mne/viz/evoked.py [hunk 4]
```diff
@@ -2046,7 +2047,9 @@ def plot_compare_evokeds(evokeds, picks=None, gfp=False, colors=None,
         callable, it must take as its single argument an array
         (observations x times) and return the upper and lower confidence bands.
         If None or False, no confidence band is plotted.
-        If True, a 95%% bootstrapped confidence interval is drawn.
+        If True, a 95%% bootstrapped confidence interval is drawn, unless
+        ``evokeds`` does not have > 2 Evoked objects for each condition,
+        in which case no CI is drawn.
```
Comment (agramfort) [inline mne/viz/evoked.py]: if there is a decision logic then I prefer to use 'auto' as if I pass True I don't expect some logic to change this to False.

File: mne/viz/evoked.py [hunk 5]
```diff
@@ -2127,6 +2133,11 @@ def plot_compare_evokeds(evokeds, picks=None, gfp=False, colors=None,
     evokeds, colors = _format_evokeds_colors(evokeds, cmap, colors)
     conditions = sorted(list(evokeds.keys()))
 
+    if ci and any(len(evokeds_) < 2 for evokeds_ in evokeds.values()):
+        if not isinstance(ci, bool):
+            warn("Only one Evoked object per condition. Not plotting the CI.")
+        ci = False
```
Comment (agramfort) [inline mne/viz/evoked.py]: once you switched to 'auto' option isolate the logic of setting ci with a private function.
Something like `_check_ci`

File: mne/viz/ica.py [hunk 6]
```diff
@@ -279,10 +279,15 @@ def plot_ica_properties(ica, inst, picks=None, axes=None, dB=True,
...
-        plotted. For example 2.5 will plot 2.5 standard deviation above/below.
+        Whether to plot standard deviation/confidence intervals in ERP/ERF and
+        spectrum plots.
+        Defaults to True, which plots one standard deviation above/below for
+        the spectrum. If set to float allows to control how many standard
+        deviations are plotted for the spectrum. For example 2.5 will plot 2.5
+        standard deviation above/below.
+        For the ERP/ERF, by default, plot the 95 percent parametric confidence
+        interval is calculated. To change this, use ``ci`` in ``ts_args`` in
+        ``image_args`` (see below).
```
Comment (agramfort) [inline mne/viz/ica.py]: why is this called ci at some place and std at others? When I read plot_std for me you report +/- 1 standard deviation on the plots. This needs to be made consistent.

Comment (mmagnuski) [reply mne/viz/ica.py]: Yes, for psd plot we first used +/-1 std but later when evoked plot was refactored to use plot_compare_evokeds it used 95% CI. So currently spectrum uses +/- 1 std and evoked plot uses CI.

**Comment by agramfort** [comment]
so anything to fix / improve for consistency / clarity?

>

File: mne/viz/evoked.py [hunk 7]
```diff
@@ -1858,7 +1858,7 @@ def _calculate_ci_and_mean(evokeds, conditions, scaling, picks, ci_fun, gfp):
     return data_dict, ci_dict
 
 
-def _get_ci_function_for_evokeds(ci):
+def _get_ci_function_for_evokeds(ci, do_topo=False):
```
Comment (jona-sassenhagen) [inline mne/viz/evoked.py]: @agramfort 

>once you switched to 'auto' option isolate the logic of setting ci with a private function.
Something like _check_ci

like this?

Also see `mne.stats._ci` (ands below 1880)

File: mne/viz/ica.py (hunk 6 above)
Comment (jona-sassenhagen) [reply mne/viz/ica.py]: I am not sure what would be best. The evoked reuses `plot_compare_evokeds`. We could use the one-sided CI functionality in both evoked and psd (and add it to `mne.stats._ci´), and call it a 95% CI for both.

**Comment by agramfort** [comment]
my suggestions are not clear @jona-sassenhagen ?

**Comment by agramfort** [comment]
before thinking about code changes, what would be a consistent API for all
functions that can display a confidence interval / uncertainty on an
average?

**Comment by larsoner** [comment]
@jona-sassenhagen do you think we can sort this out in the next couple of days to get it in for 0.18, or should we move the milestone to 0.19?

**Comment by jona-sassenhagen** [comment]
The brainvision reader is IMO more important than this one, so I don't mind moving this to .19.

**Comment by agramfort** [comment]
what's left to do here @jona-sassenhagen ?

**Comment by jona-sassenhagen** [comment]
i'd like to wait for #6422.

**Comment by jona-sassenhagen** [comment]
```
##[section]Starting: Run tests
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.151.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents:
pytest -m "not ultraslowtest" mne
========================== Starting Command Output ===========================
##[command]"C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\02d3479f-eba7-4944-9140-065b5e3d239c.cmd""
'pytest' is not recognized as an internal or external command,
operable program or batch file.
##[error]Cmd.exe exited with code '1'.
##[section]Finishing: Run tests
```

What?

**Comment by mmagnuski** [comment]
I don't know if this is related but Azure here seems to have the same problem during installation as in this other PR, see [here](https://github.com/mne-tools/mne-python/pull/6551#issuecomment-513476181_).
There it seems that fixing `CONDA_VERSION` to `4.7.5` in `azure-pipelines.yml` helped.

**Comment by GuillaumeFavelier** [comment]
I looked at the failing [Windows Python37-64bit-full-conda](https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=2354) build and at the end of the log of `Install dependencies with conda`, I see that you encounter the same error message as me:

```
LinkError: post-link script failed for package defaults::ipykernel-5.1.1-py37h39e3cac_0
location of failed script: D:\a\1\s\conda\Scripts\.ipykernel-post-link.bat
==> script messages <==
<None>
==> script output <==
stdout: 
stderr: '"D:\a\1\s\conda\condabin\conda.bat"' is not recognized as an internal or external command,

operable program or batch file.


return code: 1

()
```

**Comment by larsoner** [comment]
@jona-sassenhagen rebase and Azure should be okay. But we can also ignore it and trust Travis for this if you want

File: mne/viz/ica.py (hunk 6 above)
Comment (jona-sassenhagen) [reply mne/viz/ica.py]: @mmagnuski ?

**Comment by jona-sassenhagen** [comment]
Rebase fixed this.

Further comments @drammock @agramfort @mmagnuski ?

**Comment by drammock** [comment]
code looks reasonable. I agree with @agramfort's point about `ci` vs `std` --- it would be good to unify handling of confidence bands.  But (unless I'm missing something) I don't see that as a blocker for this PR.

**Comment by agramfort** [comment]
thx @jona-sassenhagen

**Pull request merged**